cooperator/models/subscription_request.py

```diff
@@ -508,6 +513,7 @@ def get_eater_vals(self, partner, share_product_id):  # noqa
 
     def _prepare_invoice_line(self, product, partner, qty):
         self.ensure_one()
+        product = product.with_company(self.company_id)
```

Comment:
We can assure that the company of the user environment is the same of the `subscription.request` so I'm not sure why we need to explicit call the company.
We can assure it by the `ir.rules` of the model `subscription.request`

Reply:
without this, validating a subscription request for another company than the current one (which can happen when a user is “logged in” to multiple companies) works. this is covered by `test_create_cooperator_for_non_current_company()`. sorry for the lack of comment. comment added.